types/src/cryptography/hash.rs

```diff
@@ -25,6 +26,27 @@ impl Hashable for Address {
     }
 }
 
+impl Hashable for u128 {
+    fn hash_custom(&self) -> Hash {
+        hash(self.to_be_bytes())
+    }
+}
+
+impl Hashable for Timestamp {
+    fn hash_custom(&self) -> Hash {
+        hash(self.as_micros().to_be_bytes())
+    }
+}
+
+impl Hashable for (u128, Timestamp) {
+    fn hash_custom(&self) -> Hash {
+        let mut bytes = Vec::new();
+        bytes.extend_from_slice(&self.0.to_be_bytes());
+        bytes.extend_from_slice(&self.1.as_micros().to_be_bytes());
+        hash(&bytes)
+    }
+}
```

Comment:
NIT: Heap allocation could be avoided using a 256B array (could be lazily initialized, but that would require unsafe). Or at least reserve the Vec capacity to avoid potentially allocating it twice (`Vec::with_capacity(256)`).

Reply:
Cool, I switched to an array